src/TSTransformer/nodes/statements/transformForOfStatement.ts

```diff
@@ -435,6 +437,50 @@ function getLoopBuilder(state: TransformState, node: ts.Node, type: ts.Type): Lo
 	}
 }
 
+export function transformForOfRangeMacro(state: TransformState, node: ts.ForOfStatement): luau.List<luau.Statement> {
+	// we proved this before
+	assert(ts.isCallExpression(node.expression));
+
```

Comment:
This could be done with a type in the parameter, eg:
`node: ts.ForOfStatement & { expression: ts.CallExpression }`